go/adbc/driver/bigquery/statement.go

```diff
@@ -0,0 +1,490 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package bigquery
+
+import (
+	"cloud.google.com/go/bigquery"
+	"context"
+	"fmt"
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow/go/v16/arrow"
+	"github.com/apache/arrow/go/v16/arrow/array"
+	"github.com/apache/arrow/go/v16/arrow/memory"
+	"google.golang.org/api/option"
+	"time"
+)
+
+// todos for bigquery.QueryConfig
+// - TableDefinitions
+// - Parameters
+// - TimePartitioning
+// - RangePartitioning
+// - Clustering
+// - Labels
+// - DestinationEncryptionConfig
+// - SchemaUpdateOptions
+// - ConnectionProperties
+
+type statement struct {
+	ConnectionImpl *ConnectionImpl
+	query          string
+	queryConfig    bigquery.QueryConfig
+	projectID      *string
+
+	alloc memory.Allocator
+}
+
+// Close releases any relevant resources associated with this statement
+// and closes it (particularly if it is a prepared statement).
+//
+// A statement instance should not be used after Close is called.
+func (st *statement) Close() error {
+	return nil
+}
+
+func (st *statement) GetOption(key string) (string, error) {
+	switch key {
+	case OptionStringProjectID:
+		if st.projectID == nil {
+			val, err := st.ConnectionImpl.GetOption(OptionStringProjectID)
+			if err != nil {
+				return "", err
+			} else {
+				return val, nil
+			}
+		} else {
+			return *st.projectID, nil
+		}
```

Comment:
why would the projectID of a statement be nil? shouldn't it always match it's corresponding connection?